src/server/server_family.cc

```diff
@@ -1164,10 +1164,29 @@ string GetPassword() {
   return "";
 }
 
+void ServerFamily::SendInvalidationMessages() const {
+  // send invalidation message (caused by flushdb) to all the clients which
```

Comment:
how does this work in redis if there are several dbs? 
flush db is not flush all

Reply:
> how does this work in redis if there are several dbs? flush db is not flush all

Good question, and Redis does the same. 

![image](https://github.com/dragonflydb/dragonfly/assets/61070669/add9025d-14dc-4940-80ec-fe04c7cc42e7)

This behavior actually was in Redis' spec of client side caching, and the reason is simply about lowering implementation complexity:

[https://redis.io/docs/manual/client-side-caching/](https://www.google.com/url?q=https://redis.io/docs/manual/client-side-caching/&sa=D&source=docs&ust=1702327521483711&usg=AOvVaw0QFAUH76qBoZ5hALTYZ_BU)

"There is a single keys namespace, not divided by database numbers. So if a client is caching the key foo in database 2, and some other client changes the value of the key foo in database 3, an invalidation message will still be sent. This way we can ignore database numbers reducing both the memory usage and the implementation complexity."